### DISCUSSION TITLE
Local Resource Full Support

### DISCUSSION THREAD

**Discussion by rohitsanj** [confluentinc/vscode#454] [Ideas]
### Users

- [ ] Development manager tasked with ensuring a team is successful in using streaming data
- [ ] Developer on a team tasked with building new features using streaming data
- [ ] Developer at an org who manages a streaming platform while building new features that use streaming data
- [ ] Developer tasked with building pipelines for use in business analytics
- [ ] Other: [User type] tasked with [primary goal in the realm of streaming data]

### Problems

1.
2.
3.

### Solutions

1.
2.
3.

### Goals

1.
2.
3.

### Non-Goals

1.
2.
3.

### Background

TODO

### Proposal

TODO

**Comment by derek1ee** [comment]
Proposal:

1. The `RESOURCES` section of the tree shall have two L1 items: `Confluent Cloud` and `Direct Connect`.
1. No change to `Confluent Cloud` portion of the tree, L2 shows environments, L3 shows clusters / schema registry.
1. The `Direct Connect` shall display the following items on L2: `Local Sandbox` and direct connection(s).
   1. The `Sandbox` option will always be shown (cannot be deleted). It can be started/stoped/stopped w/ data purged.
      1. By default the extension will start/stop all available services (Kafka, schema registry, etc.) on free ports.
      1. User shall have the ability to start/stop individual service via command palette.
      1. When starting SR separately, there shall be an optional parameter for the Kafka bootstrap server user can specify, if they want to tie the schema registry to a non-Sandbox kafka cluster (e.g. one started via CLI `confluent local kafka start`)
   1. A `Local` option will always be shown (cannot be deleted), it will display services running locally, started outside of the extension.
   1. 0 or more `Connection` items are shown below `Local`, and it will contain information for 0 or more of [kafka, schema registry, etc.]
      1. User can create/edit (e.g. update API key, update a connection from just kafka to kafka + SR)/delete connection.
      1. A connection will support local/remote/CP.
   1. We shall leverage iconography (e.g. beaker vs computer vs globe), available actions (start/stop vs. edit connection), tooltips to help user understand and differentiate the different entities.

![image](https://github.com/user-attachments/assets/b742c74a-d3be-4336-9bfd-657a4debad55)

**Comment by rohitsanj** [comment]
We have this now.
